codify/deployment.go

```diff
@@ -28,36 +28,40 @@ import (
 	"fmt"
 	"github.com/kris-nova/logger"
 	appsv1 "k8s.io/api/apps/v1"
+	"strings"
 	"text/template"
 	"time"
 )
 
 type Deployment struct {
-	i *appsv1.Deployment
+	*appsv1.Deployment
+	GoName string
 }
 
 func NewDeployment(obj *appsv1.Deployment) *Deployment {
 	obj.ObjectMeta = cleanObjectMeta(obj.ObjectMeta)
 	obj.Status = appsv1.DeploymentStatus{}
 	return &Deployment{
-		i: obj,
+		Deployment: obj,
+		GoName:     strings.ReplaceAll(obj.Name, "-", "_"),
 	}
 }
 
 func (k Deployment) Install() string {
-	l := Literal(k.i)
+	l := Literal(k.Deployment)
 	install := fmt.Sprintf(`
-	{{ .Name }}Deployment := %s
+	// Adding a deployment: "{{ .Name }}"
+	{{ .GoName }}Deployment := %s
 
-	_, err = client.AppsV1().Deployments("{{ .Namespace }}").Create(context.TODO(), {{ .Name }}Deployment, v1.CreateOptions{})
+	_, err = client.AppsV1().Deployments("{{ .Namespace }}").Create(context.TODO(), {{ .GoName }}Deployment, v1.CreateOptions{})
```

Comment:
GoName is an amazing idea - i’m glad you are giving these files some love - it took about 10 tries of hacking crap together to finally get the code generation this stable - great idea 🎉